install/swiftly-install.sh

```diff
@@ -71,9 +115,57 @@ bold () {
     echo "$(tput bold)$1$(tput sgr0)"
 }
 
+# Fetch the list of required system dependencies from the apple/swift-docker
+# repository and attempt to install them using the system's package manager.
+#
+# $docker_platform_name, $docker_platform_version, and $package manager need
+# to be set before calling this function.
+install_system_deps () {
+    if [[ "$(id --user)" != "0" ]] && ! has_command sudo ; then
+        echo "Warning: sudo not installed and current user is not root, skipping system dependency installation."
+        return
+    elif ! has_command "$package_manager" ; then
+        echo "Warning: package manager \"$package_manager\" not found, skipping system dependency instllation."
+        return
+    fi
+
+    dockerfile_url="https://raw.githubusercontent.com/apple/swift-docker/main/nightly-main/$docker_platform_name/$docker_platform_version/Dockerfile"
+    dockerfile="$(curl --silent --retry 3 --location --fail $dockerfile_url)"
+    if [[ "$?" -ne 0 ]]; then
+        echo "Error enumerating system dependencies, skipping installation of system dependencies."
+    fi
+
+    # Find the line number of the RUN command associated with installing system dependencies.
+    beg_line_num=$(printf "$dockerfile" | grep -n --max-count=1 "$package_manager.*install" | cut -d ":" -f1)
+
+    # Starting from there, find the first line that doesn't have the same level of indentation.
```

Comment:
Indentation seems a little flaky way to catch the full list of installation dependencies. Why don't you use the fact that the last non-whitespace character is a backslash. Maybe once you have found the install line concatenate all the lines ending with `\` together. Then split the line into separate commands ie with `&&` as separator. Then choose the install command and read the packages from it.

Its still not ideal and just hardcoding the required packages might just be preferable.

If you are going to parse the Docker file then I'd also add tests for the parsing.

Reply:
Yeah, indentation might not be be the best thing to rely on. Updated to do something more along the lines of your suggestion. I don't really want to hardcode any lists of dependencies since they vary from platform to platform and are updated every now and then.

The tests (in particular default-install.sh) do cover the parsing behavior, though not as comprehensively as they should. I've updated default-install.sh to verify more of the required dependencies were installed. For that I just hardcoded the lists, which is okay since they're just tests. We may need to update it from time to time, but the cost of doing so shouldn't be too bad.

Reply:
My regex, linux commands knowledge is a little rusty so I'm gonna have to trust you on this. But much prefer checking for `\` than the tabbing.